userProfile.php

```diff
@@ -0,0 +1,39 @@
+<?php
+use Icepace\UserHydrator;
+
+require_once "vendor/autoload.php";
+
+$username = $_GET['username'];
+$db = new PDO('mysql:host=db; dbname=icepace', 'root', 'password');
+$user = UserHydrator::getUserByUsername($db, $username);
```

Comment:
there is no logic here to handle the case where the user inputs an invalid username into the URL:
![image](https://user-images.githubusercontent.com/25844355/190124959-e467167c-0be4-447a-adfe-6556137ffd78.png)

Reply:
Logic added - try and catch block that if it throws a TypeError it re-directs you back to index.php.